### PR TITLE
Create banner for authors campaign / May 2018

### PR DESCRIPTION
**Pull request by ghost**
Hey guys, this banner is supposed to go live on Monday, May 7th at noon. Would be cool if you could provide some quick feedback in the morning on the code. 

The original ticket suggested taking some existing banner and slapping the new design onto it, but since the old banner was kind of bad in the CSS department I had issues getting the new design to work / look properly, so I ended up basically creating the banner almost from scratch with a more modern flex-box-based CSS approach. For maintainability purposes, we also ended up creating this repository so that future campaign banners can be tracked more easily.

### CODE REVIEW

**Comment by ghost** [comment]
@gbirke In addition to the PR, could you make `master` the default branch and delete `init` afterwards?

File: authors/AuthorsBanner.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	createBanner() {
+		const $bannerContainer = $( '#WMDE-Banner-Container' );
+		$bannerContainer.html( this.bannerTemplate( {} ) );
+		this.addBannerSpace();
+		$( 'body' ).prepend( $( '#centralNotice' ) );
+		$( '#author-banner-container' ).show();
+	}
+
+	removeBanner() {
+		$( '#author-banner-container' ).hide();
```
Comment (gbirke) [inline authors/AuthorsBanner.js]: Why is this ID different from `#WMDE-Banner-Container`? Shouldn't the container be stored in the initialization and reused in the methods?

File: authors/AuthorsBanner.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		$( '#ca-ve-edit, .mw-editsection-visualeditor' ).click( function ( ) {
+			$( '#author-banner-container' ).hide();
+			bannerInstance.removeBanner();
+		} );
+	}
+
+	addBannerSpace() {
+		const bannerHeight = $( '#author-banner-container' ).height() + 20;
+		switch ( mw.config.get( 'skin' ) ) {
+			case 'vector': {
```
Comment (gbirke) [inline authors/AuthorsBanner.js:52]: are the curly brackets for the cases really necessary? Never seen something like that.

File: authors/AuthorsBanner.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+			}
+			case 'monobook': {
+				const wrapper = $( '#globalWrapper' );
+				const banner = $( '#author-banner' );
+				wrapper.css( 'position', 'relative' );
+				wrapper.css( 'top', bannerHeight );
+				banner.css( 'top', '-20px' );
+				banner.css( 'background', 'none' );
+				break;
+			}
```
Comment (gbirke) [inline authors/AuthorsBanner.js:66]: On the timless skin the banner obfuscates part of the page. Maybe we should ask which skins are supported and what to do when encountering unsupported skins.

**Review by tzhelyazkova — COMMENTED** [review]
the js linter complains
```
> authors-banners-webpack@1.0.0 lint:js /code
> eslint --cache --cache-location .eslintcache --ignore-path .gitignore .
/code/authors/AuthorsBanner.js
  44:68  error  There should be no spaces inside this paren  space-in-parens

/code/authors/banner_ctrl.js
  9:8  error  'authorsBanner' is assigned a value but never used  no-unused-vars

/code/authors/banner_var.js
  9:8  error  'authorsBanner' is assigned a value but never used  no-unused-vars

✖ 3 problems (3 errors, 0 warnings)
  1 error, 0 warnings potentially fixable with the `--fix` option.
```

**Review by gbirke — APPROVED** [review]
Works as defined in the issue

File: authors/AuthorsBanner.js (hunk 2 above)
Comment (ghost) [reply authors/AuthorsBanner.js:52]: See https://eslint.org/docs/rules/no-case-declarations

**Pull request merged**